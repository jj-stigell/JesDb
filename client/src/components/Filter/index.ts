export { default, initFilters } from './Filter'
