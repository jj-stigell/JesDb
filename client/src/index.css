@tailwind base;
@tailwind components;
@tailwind utilities;

@layer components {
  .link {
    @apply underline mx-1 text-blue-600 hover:text-blue-800 dark:text-blue-300 dark:hover:text-blue-600;
  }
  .title {
    @apply my-2 text-xl text-black font-extrabold dark:text-white;
  }
}

body {
  margin: 0;
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', 'Roboto', 'Oxygen',
    'Ubuntu', 'Cantarell', 'Fira Sans', 'Droid Sans', 'Helvetica Neue',
    sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

code {
  font-family: source-code-pro, Menlo, Monaco, Consolas, 'Courier New',
    monospace;
}

.keyword {
  color: red;
}
