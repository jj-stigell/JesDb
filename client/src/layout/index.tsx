import React from 'react'
import { Outlet } from 'react-router-dom'

import Modal from './Modal'
import Menu from './Menu'
import Footer from './Footer'

export default function MainLayout (): React.JSX.Element {
  return (
    <>
      <div
        className="dark:bg-jesdark-900"
        style={{ minHeight: '100vh', display: 'flex', flexDirection: 'column' }}
      >
        <Modal />
        <Menu />
        <Outlet/>
      </div>
      <Footer />
    </>
  )
}
