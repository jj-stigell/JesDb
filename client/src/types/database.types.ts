// DO NOT EDIT THIS FILE
// TYPES ARE AUTOGENERATED FROM DATABASE SCHEMA

export type Json =
  | string
  | number
  | boolean
  | null
  | { [key: string]: Json | undefined }
  | Json[]

export interface Database {
  public: {
    Tables: {
      notes: {
        Row: {
          audio: string | null
          created_at: string
          external_id: number
          id: number
          image: string | null
          sentence: string
          sentence_furigana: string
          source: Database['public']['Enums']['media_source'] | null
          source_id: string | null
          start_time: number | null
          translation: string | null
        }
        Insert: {
          audio?: string | null
          created_at?: string
          external_id: number
          id?: number
          image?: string | null
          sentence: string
          sentence_furigana: string
          source?: Database['public']['Enums']['media_source'] | null
          source_id?: string | null
          start_time?: number | null
          translation?: string | null
        }
        Update: {
          audio?: string | null
          created_at?: string
          external_id?: number
          id?: number
          image?: string | null
          sentence?: string
          sentence_furigana?: string
          source?: Database['public']['Enums']['media_source'] | null
          source_id?: string | null
          start_time?: number | null
          translation?: string | null
        }
        Relationships: []
      }
      profiles: {
        Row: {
          action_type: Database['public']['Enums']['action_type']
          audio_field: string | null
          created_at: string
          deck_name: string
          expression_field: string | null
          id: number
          image_field: string | null
          name: string
          note_type: string
          reading_field: string | null
          translation_field: string | null
          user_id: string
        }
        Insert: {
          action_type: Database['public']['Enums']['action_type']
          audio_field?: string | null
          created_at?: string
          deck_name: string
          expression_field?: string | null
          id?: number
          image_field?: string | null
          name: string
          note_type: string
          reading_field?: string | null
          translation_field?: string | null
          user_id: string
        }
        Update: {
          action_type?: Database['public']['Enums']['action_type']
          audio_field?: string | null
          created_at?: string
          deck_name?: string
          expression_field?: string | null
          id?: number
          image_field?: string | null
          name?: string
          note_type?: string
          reading_field?: string | null
          translation_field?: string | null
          user_id?: string
        }
        Relationships: [
          {
            foreignKeyName: 'public_profile_user_id_fkey'
            columns: ['user_id']
            isOneToOne: false
            referencedRelation: 'users'
            referencedColumns: ['id']
          },
        ]
      }
      synonyms: {
        Row: {
          synonyms: string[] | null
          term: string
        }
        Insert: {
          synonyms?: string[] | null
          term: string
        }
        Update: {
          synonyms?: string[] | null
          term?: string
        }
        Relationships: []
      }
    }
    Views: {
      [_ in never]: never
    }
    Functions: {
      delete_user: {
        Args: Record<PropertyKey, never>
        Returns: undefined
      }
      search_sentences: {
        Args: {
          keyword: string
        }
        Returns: Array<{
          id: number
          sentence: string
          len: number
          sentence_furigana: string
          translation: string
          audio: string
          image: string
          start_time: number
          source_id: string
          highlighted_sentence: string
          highlighted_language: string
          sentence_furigana_highlight: string
        }>
      }
    }
    Enums: {
      action_type: 'addNote' | 'updateNote'
      media_source: 'youtube'
    }
    CompositeTypes: {
      [_ in never]: never
    }
  }
}

type PublicSchema = Database[Extract<keyof Database, 'public'>]

export type Tables<
  PublicTableNameOrOptions extends
  | keyof (PublicSchema['Tables'] & PublicSchema['Views'])
  | { schema: keyof Database },
  TableName extends PublicTableNameOrOptions extends { schema: keyof Database }
    ? keyof (Database[PublicTableNameOrOptions['schema']]['Tables'] &
    Database[PublicTableNameOrOptions['schema']]['Views'])
    : never = never,
> = PublicTableNameOrOptions extends { schema: keyof Database }
  ? (Database[PublicTableNameOrOptions['schema']]['Tables'] &
    Database[PublicTableNameOrOptions['schema']]['Views'])[TableName] extends {
      Row: infer R
    }
      ? R
      : never
  : PublicTableNameOrOptions extends keyof (PublicSchema['Tables'] &
  PublicSchema['Views'])
    ? (PublicSchema['Tables'] &
      PublicSchema['Views'])[PublicTableNameOrOptions] extends {
        Row: infer R
      }
        ? R
        : never
    : never

export type TablesInsert<
  PublicTableNameOrOptions extends
  | keyof PublicSchema['Tables']
  | { schema: keyof Database },
  TableName extends PublicTableNameOrOptions extends { schema: keyof Database }
    ? keyof Database[PublicTableNameOrOptions['schema']]['Tables']
    : never = never,
> = PublicTableNameOrOptions extends { schema: keyof Database }
  ? Database[PublicTableNameOrOptions['schema']]['Tables'][TableName] extends {
    Insert: infer I
  }
    ? I
    : never
  : PublicTableNameOrOptions extends keyof PublicSchema['Tables']
    ? PublicSchema['Tables'][PublicTableNameOrOptions] extends {
      Insert: infer I
    }
      ? I
      : never
    : never

export type TablesUpdate<
  PublicTableNameOrOptions extends
  | keyof PublicSchema['Tables']
  | { schema: keyof Database },
  TableName extends PublicTableNameOrOptions extends { schema: keyof Database }
    ? keyof Database[PublicTableNameOrOptions['schema']]['Tables']
    : never = never,
> = PublicTableNameOrOptions extends { schema: keyof Database }
  ? Database[PublicTableNameOrOptions['schema']]['Tables'][TableName] extends {
    Update: infer U
  }
    ? U
    : never
  : PublicTableNameOrOptions extends keyof PublicSchema['Tables']
    ? PublicSchema['Tables'][PublicTableNameOrOptions] extends {
      Update: infer U
    }
      ? U
      : never
    : never

export type Enums<
  PublicEnumNameOrOptions extends
  | keyof PublicSchema['Enums']
  | { schema: keyof Database },
  EnumName extends PublicEnumNameOrOptions extends { schema: keyof Database }
    ? keyof Database[PublicEnumNameOrOptions['schema']]['Enums']
    : never = never,
> = PublicEnumNameOrOptions extends { schema: keyof Database }
  ? Database[PublicEnumNameOrOptions['schema']]['Enums'][EnumName]
  : PublicEnumNameOrOptions extends keyof PublicSchema['Enums']
    ? PublicSchema['Enums'][PublicEnumNameOrOptions]
    : never
