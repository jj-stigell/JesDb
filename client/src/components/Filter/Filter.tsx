import React, { FormEvent } from 'react'
import { Label, Radio } from 'flowbite-react'

import { Filters } from 'src/types'

export const initFilters: Filters = {
  sort: 'relevance',
  onlyTranslated: false
}

interface IFilter {
  setFilters: (filters: Filters) => void
  filters: Filters
  disable: boolean
}

export default function Filter ({
  setFilters,
  filters,
  disable
}: IFilter): React.JSX.Element {
  const setSort = (event: FormEvent<HTMLFieldSetElement>): void => {
    const target = event.target as HTMLInputElement
    const value = target.value as Filters['sort']
    setFilters({ ...filters, sort: value })
  }

  const setTranslation = (event: FormEvent<HTMLFieldSetElement>): void => {
    const target = event.target as HTMLInputElement
    const value = target.value
    setFilters({ ...filters, onlyTranslated: value === 'translation-required' })
  }

  const resetFilters = (): void => {
    setFilters(initFilters)
  }

  return (
    <div
      className={`
        mb-2 border-x-2 border-b-2 rounded-lg border-blue-100
        p-2 mt-1 dark:border-jesdark-800 dark:bg-jesdark-600
      `}
    >
      <div className="flex justify-end">
        <button
          onClick={resetFilters}
          disabled={disable}
          className="underline text-red-400 hover:text-red-500"
          aria-label="Reset filters"
        >
          Reset filters
        </button>
      </div>
      <div className="flex items-center space-x-2">
        <div className="w-3/6">
          <fieldset
            onChange={setSort}
            className="flex max-w-md flex-rows flex-wrap gap-4"
            aria-labelledby="sort-by-legend"
          >
            <legend id="sort-by-legend" className="mb-2 dark:text-white">
              Sort by
            </legend>
            <div className="flex items-center gap-1">
              <Radio
                id="relevance"
                name="sort"
                value="relevance"
                checked={filters.sort === 'relevance'}
                onChange={() => {}}
                disabled={disable}
              />
              <Label htmlFor="relevance">Relevance</Label>
            </div>
            <div className="flex items-center gap-1">
              <Radio
                id="shortness"
                name="sort"
                value="shortness"
                checked={filters.sort === 'shortness'}
                onChange={() => {}}
                disabled={disable}
              />
              <Label htmlFor="shortness">Shortness</Label>
            </div>
            <div className="flex items-center gap-1">
              <Radio
                id="longness"
                name="sort"
                value="longness"
                checked={filters.sort === 'longness'}
                onChange={() => {}}
                disabled={disable}
              />
              <Label htmlFor="longness">Longness</Label>
            </div>
          </fieldset>
        </div>
        <div className="w-3/6">
          <fieldset
            onChange={setTranslation}
            className="flex max-w-md flex-rows flex-wrap gap-4"
            aria-labelledby="translation-legend"
          >
            <legend id="translation-legend" className="mb-2 dark:text-white">
              Only examples with translation
            </legend>
            <div className="flex items-center gap-1">
              <Radio
                id="translation-required"
                name="translation"
                value="translation-required"
                checked={filters.onlyTranslated}
                onChange={() => {}}
                disabled={disable}
                aria-checked={filters.onlyTranslated}
              />
              <Label htmlFor="translation-required">Yes</Label>
            </div>
            <div className="flex items-center gap-1">
              <Radio
                id="translation-optional"
                name="translation"
                value="translation-optional"
                checked={!filters.onlyTranslated}
                onChange={() => {}}
                disabled={disable}
                aria-checked={!filters.onlyTranslated}
              />
              <Label htmlFor="translation-optional">No</Label>
            </div>
          </fieldset>
        </div>
      </div>
    </div>
  )
}
