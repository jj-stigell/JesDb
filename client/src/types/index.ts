import { Database } from './database.types'

/**
 * No params for deckNames, modelNamesAndIds, modelNames
 */
export type Action =
  'deckNames' |
  'updateNote' |
  'modelNamesAndIds' |
  'modelFieldNames' |
  'modelNames' |
  'findNotes' |
  'addNote' |
  'getNoteTags'

export interface UpdateNoteParams {
  note: {
    id: number
    fields: Record<string, string>
    tags: string[]
  }
}

export interface AddNoteParams {
  note: {
    deckName: string
    modelName: string
    fields: Record<string, string>
    options?: {
      allowDuplicate?: boolean
      duplicateScope?: string
      duplicateScopeOptions?: {
        deckName?: string
        checkChildren?: boolean
        checkAllModels?: boolean
      }
    }
    tags?: string[]
    audio?: Array<{
      url: string
      filename: string
      skipHash: string
      fields: string[]
    }>
    video?: Array<{
      url: string
      filename: string
      skipHash: string
      fields: string[]
    }>
    picture?: Array<{
      url: string
      filename: string
      skipHash: string
      fields: string[]
    }>
  }
}

export type NoteParams = UpdateNoteParams | AddNoteParams

interface ModelFieldNamesParams {
  modelName: string
}

interface FindNotesParams {
  query: string
}

interface GetNoteTagsParams {
  note: number
}

interface ActionParamsMap {
  deckNames: undefined
  updateNote: UpdateNoteParams
  modelNamesAndIds: undefined
  modelFieldNames: ModelFieldNamesParams
  modelNames: undefined
  findNotes: FindNotesParams
  addNote: AddNoteParams
  getNoteTags: GetNoteTagsParams
}

export interface ErrorResult {
  result: null
  error: string
}

interface ResultStringArray {
  result: string[]
  error: null
}

export interface FindNotes {
  result: number[]
  error: null
}

export interface AddNote {
  result: number
  error: null
}

export interface UpdateNoteResult {
  result: null
  error: null
}

interface ModelNamesAndIdsResult {
  result: Array<Record<string, number>>
  error: null
}

interface ActionResultMap {
  deckNames: ResultStringArray
  updateNote: UpdateNoteResult
  modelNamesAndIds: ModelNamesAndIdsResult
  modelFieldNames: ResultStringArray
  modelNames: ResultStringArray
  findNotes: FindNotes
  addNote: AddNote
  getNoteTags: ResultStringArray
}

export type ExecuteActionParams<T extends Action> =
  T extends keyof ActionParamsMap ? ActionParamsMap[T] : never

export type ExecuteActionReturnType<T extends Action> = ActionResultMap[T]

export interface Body<T extends Action> {
  action: Action
  version: number
  params?: ExecuteActionParams<T>
}

// Represents the note type that search function returns
export type Note = Database['public']['Functions']['search_sentences']['Returns'][0]

export type Profile = Database['public']['Tables']['profiles']['Row']

export type ActionType = Database['public']['Enums']['action_type']

export interface NewProfile extends Omit<Profile, 'id' | 'user_id' | 'created_at'> {}

export interface ExportResult {
  result: string | null
  error: string | null
}

export interface Filters {
  sort: 'relevance' | 'shortness' | 'longness'
  onlyTranslated: boolean
}
