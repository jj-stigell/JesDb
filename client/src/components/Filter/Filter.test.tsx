import React from 'react'
import { render, screen, fireEvent } from '@testing-library/react'
import '@testing-library/jest-dom'

import Filter, { initFilters } from './Filter'
import { Filters } from 'src/types'

describe('Filter Component', () => {
  let setFilters: jest.Mock
  let filters: Filters

  beforeEach(() => {
    setFilters = jest.fn()
    filters = initFilters
  })

  it('renders without crashing and displays all filter options', () => {
    const { asFragment } =
      render(<Filter setFilters={setFilters} filters={filters} disable={false} />)

    expect(asFragment()).toMatchSnapshot()
    expect(screen.getByText(/Sort by/i)).toBeInTheDocument()
    expect(screen.getByText(/Only examples with translation/i)).toBeInTheDocument()
  })

  it('calls setFilters with the correct sort value when a sort option is selected', () => {
    render(<Filter setFilters={setFilters} filters={filters} disable={false} />)
    const shortnessRadio = screen.getByLabelText(/Shortness/i)
    fireEvent.click(shortnessRadio)

    expect(setFilters).toHaveBeenCalledWith({ ...filters, sort: 'shortness' })
  })

  it(
    'calls setFilters with the correct translation value when a translation option is selected',
    () => {
      render(<Filter setFilters={setFilters} filters={filters} disable={false} />)
      const translationRequiredRadio = screen.getByLabelText(/Yes/i)
      fireEvent.click(translationRequiredRadio)

      expect(setFilters).toHaveBeenCalledWith({ ...filters, onlyTranslated: true })
    })

  it('resets filters to initial values when reset filters button is clicked', () => {
    render(<Filter setFilters={setFilters} filters={filters} disable={false} />)
    const clearButton = screen.getByText(/reset filters/i)
    fireEvent.click(clearButton)

    expect(setFilters).toHaveBeenCalledWith(initFilters)
  })

  it('disables all filter options when disable prop is true', () => {
    const { asFragment } =
      render(<Filter setFilters={setFilters} filters={filters} disable={true} />)

    const relevanceRadio = screen.getByLabelText(/Relevance/i)
    const shortnessRadio = screen.getByLabelText(/Shortness/i)
    const longnessRadio = screen.getByLabelText(/Longness/i)
    const translationRequiredRadio = screen.getByLabelText(/Yes/i)
    const translationOptionalRadio = screen.getByLabelText(/No/i)

    expect(relevanceRadio).toBeDisabled()
    expect(shortnessRadio).toBeDisabled()
    expect(longnessRadio).toBeDisabled()
    expect(translationRequiredRadio).toBeDisabled()
    expect(translationOptionalRadio).toBeDisabled()
    expect(asFragment()).toMatchSnapshot()
  })

  it('matches snapshot when filters are changed', () => {
    const customFilters: Filters = {
      sort: 'longness',
      onlyTranslated: true
    }

    const { asFragment } =
      render(<Filter setFilters={setFilters} filters={customFilters} disable={false} />)

    expect(asFragment()).toMatchSnapshot()
  })
})
