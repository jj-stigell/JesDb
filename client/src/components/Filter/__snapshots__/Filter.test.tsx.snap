// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter Component disables all filter options when disable prop is true 1`] = `
<DocumentFragment>
  <div
    class="
        mb-2 border-x-2 border-b-2 rounded-lg border-blue-100
        p-2 mt-1 dark:border-jesdark-800 dark:bg-jesdark-600
      "
  >
    <div
      class="flex justify-end"
    >
      <button
        aria-label="Reset filters"
        class="underline text-red-400 hover:text-red-500"
        disabled=""
      >
        Reset filters
      </button>
    </div>
    <div
      class="flex items-center space-x-2"
    >
      <div
        class="w-3/6"
      >
        <fieldset
          aria-labelledby="sort-by-legend"
          class="flex max-w-md flex-rows flex-wrap gap-4"
        >
          <legend
            class="mb-2 dark:text-white"
            id="sort-by-legend"
          >
            Sort by
          </legend>
          <div
            class="flex items-center gap-1"
          >
            <input
              checked=""
              class="h-4 w-4 border border-gray-300 text-cyan-600 focus:ring-2 focus:ring-cyan-500 dark:border-gray-600 dark:bg-gray-700 dark:focus:bg-cyan-600 dark:focus:ring-cyan-600"
              disabled=""
              id="relevance"
              name="sort"
              type="radio"
              value="relevance"
            />
            <label
              class="text-sm font-medium text-gray-900 dark:text-white"
              data-testid="flowbite-label"
              for="relevance"
            >
              Relevance
            </label>
          </div>
          <div
            class="flex items-center gap-1"
          >
            <input
              class="h-4 w-4 border border-gray-300 text-cyan-600 focus:ring-2 focus:ring-cyan-500 dark:border-gray-600 dark:bg-gray-700 dark:focus:bg-cyan-600 dark:focus:ring-cyan-600"
              disabled=""
              id="shortness"
              name="sort"
              type="radio"
              value="shortness"
            />
            <label
              class="text-sm font-medium text-gray-900 dark:text-white"
              data-testid="flowbite-label"
              for="shortness"
            >
              Shortness
            </label>
          </div>
          <div
            class="flex items-center gap-1"
          >
            <input
              class="h-4 w-4 border border-gray-300 text-cyan-600 focus:ring-2 focus:ring-cyan-500 dark:border-gray-600 dark:bg-gray-700 dark:focus:bg-cyan-600 dark:focus:ring-cyan-600"
              disabled=""
              id="longness"
              name="sort"
              type="radio"
              value="longness"
            />
            <label
              class="text-sm font-medium text-gray-900 dark:text-white"
              data-testid="flowbite-label"
              for="longness"
            >
              Longness
            </label>
          </div>
        </fieldset>
      </div>
      <div
        class="w-3/6"
      >
        <fieldset
          aria-labelledby="translation-legend"
          class="flex max-w-md flex-rows flex-wrap gap-4"
        >
          <legend
            class="mb-2 dark:text-white"
            id="translation-legend"
          >
            Only examples with translation
          </legend>
          <div
            class="flex items-center gap-1"
          >
            <input
              aria-checked="false"
              class="h-4 w-4 border border-gray-300 text-cyan-600 focus:ring-2 focus:ring-cyan-500 dark:border-gray-600 dark:bg-gray-700 dark:focus:bg-cyan-600 dark:focus:ring-cyan-600"
              disabled=""
              id="translation-required"
              name="translation"
              type="radio"
              value="translation-required"
            />
            <label
              class="text-sm font-medium text-gray-900 dark:text-white"
              data-testid="flowbite-label"
              for="translation-required"
            >
              Yes
            </label>
          </div>
          <div
            class="flex items-center gap-1"
          >
            <input
              aria-checked="true"
              checked=""
              class="h-4 w-4 border border-gray-300 text-cyan-600 focus:ring-2 focus:ring-cyan-500 dark:border-gray-600 dark:bg-gray-700 dark:focus:bg-cyan-600 dark:focus:ring-cyan-600"
              disabled=""
              id="translation-optional"
              name="translation"
              type="radio"
              value="translation-optional"
            />
            <label
              class="text-sm font-medium text-gray-900 dark:text-white"
              data-testid="flowbite-label"
              for="translation-optional"
            >
              No
            </label>
          </div>
        </fieldset>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Filter Component matches snapshot when filters are changed 1`] = `
<DocumentFragment>
  <div
    class="
        mb-2 border-x-2 border-b-2 rounded-lg border-blue-100
        p-2 mt-1 dark:border-jesdark-800 dark:bg-jesdark-600
      "
  >
    <div
      class="flex justify-end"
    >
      <button
        aria-label="Reset filters"
        class="underline text-red-400 hover:text-red-500"
      >
        Reset filters
      </button>
    </div>
    <div
      class="flex items-center space-x-2"
    >
      <div
        class="w-3/6"
      >
        <fieldset
          aria-labelledby="sort-by-legend"
          class="flex max-w-md flex-rows flex-wrap gap-4"
        >
          <legend
            class="mb-2 dark:text-white"
            id="sort-by-legend"
          >
            Sort by
          </legend>
          <div
            class="flex items-center gap-1"
          >
            <input
              class="h-4 w-4 border border-gray-300 text-cyan-600 focus:ring-2 focus:ring-cyan-500 dark:border-gray-600 dark:bg-gray-700 dark:focus:bg-cyan-600 dark:focus:ring-cyan-600"
              id="relevance"
              name="sort"
              type="radio"
              value="relevance"
            />
            <label
              class="text-sm font-medium text-gray-900 dark:text-white"
              data-testid="flowbite-label"
              for="relevance"
            >
              Relevance
            </label>
          </div>
          <div
            class="flex items-center gap-1"
          >
            <input
              class="h-4 w-4 border border-gray-300 text-cyan-600 focus:ring-2 focus:ring-cyan-500 dark:border-gray-600 dark:bg-gray-700 dark:focus:bg-cyan-600 dark:focus:ring-cyan-600"
              id="shortness"
              name="sort"
              type="radio"
              value="shortness"
            />
            <label
              class="text-sm font-medium text-gray-900 dark:text-white"
              data-testid="flowbite-label"
              for="shortness"
            >
              Shortness
            </label>
          </div>
          <div
            class="flex items-center gap-1"
          >
            <input
              checked=""
              class="h-4 w-4 border border-gray-300 text-cyan-600 focus:ring-2 focus:ring-cyan-500 dark:border-gray-600 dark:bg-gray-700 dark:focus:bg-cyan-600 dark:focus:ring-cyan-600"
              id="longness"
              name="sort"
              type="radio"
              value="longness"
            />
            <label
              class="text-sm font-medium text-gray-900 dark:text-white"
              data-testid="flowbite-label"
              for="longness"
            >
              Longness
            </label>
          </div>
        </fieldset>
      </div>
      <div
        class="w-3/6"
      >
        <fieldset
          aria-labelledby="translation-legend"
          class="flex max-w-md flex-rows flex-wrap gap-4"
        >
          <legend
            class="mb-2 dark:text-white"
            id="translation-legend"
          >
            Only examples with translation
          </legend>
          <div
            class="flex items-center gap-1"
          >
            <input
              aria-checked="true"
              checked=""
              class="h-4 w-4 border border-gray-300 text-cyan-600 focus:ring-2 focus:ring-cyan-500 dark:border-gray-600 dark:bg-gray-700 dark:focus:bg-cyan-600 dark:focus:ring-cyan-600"
              id="translation-required"
              name="translation"
              type="radio"
              value="translation-required"
            />
            <label
              class="text-sm font-medium text-gray-900 dark:text-white"
              data-testid="flowbite-label"
              for="translation-required"
            >
              Yes
            </label>
          </div>
          <div
            class="flex items-center gap-1"
          >
            <input
              aria-checked="false"
              class="h-4 w-4 border border-gray-300 text-cyan-600 focus:ring-2 focus:ring-cyan-500 dark:border-gray-600 dark:bg-gray-700 dark:focus:bg-cyan-600 dark:focus:ring-cyan-600"
              id="translation-optional"
              name="translation"
              type="radio"
              value="translation-optional"
            />
            <label
              class="text-sm font-medium text-gray-900 dark:text-white"
              data-testid="flowbite-label"
              for="translation-optional"
            >
              No
            </label>
          </div>
        </fieldset>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Filter Component renders without crashing and displays all filter options 1`] = `
<DocumentFragment>
  <div
    class="
        mb-2 border-x-2 border-b-2 rounded-lg border-blue-100
        p-2 mt-1 dark:border-jesdark-800 dark:bg-jesdark-600
      "
  >
    <div
      class="flex justify-end"
    >
      <button
        aria-label="Reset filters"
        class="underline text-red-400 hover:text-red-500"
      >
        Reset filters
      </button>
    </div>
    <div
      class="flex items-center space-x-2"
    >
      <div
        class="w-3/6"
      >
        <fieldset
          aria-labelledby="sort-by-legend"
          class="flex max-w-md flex-rows flex-wrap gap-4"
        >
          <legend
            class="mb-2 dark:text-white"
            id="sort-by-legend"
          >
            Sort by
          </legend>
          <div
            class="flex items-center gap-1"
          >
            <input
              checked=""
              class="h-4 w-4 border border-gray-300 text-cyan-600 focus:ring-2 focus:ring-cyan-500 dark:border-gray-600 dark:bg-gray-700 dark:focus:bg-cyan-600 dark:focus:ring-cyan-600"
              id="relevance"
              name="sort"
              type="radio"
              value="relevance"
            />
            <label
              class="text-sm font-medium text-gray-900 dark:text-white"
              data-testid="flowbite-label"
              for="relevance"
            >
              Relevance
            </label>
          </div>
          <div
            class="flex items-center gap-1"
          >
            <input
              class="h-4 w-4 border border-gray-300 text-cyan-600 focus:ring-2 focus:ring-cyan-500 dark:border-gray-600 dark:bg-gray-700 dark:focus:bg-cyan-600 dark:focus:ring-cyan-600"
              id="shortness"
              name="sort"
              type="radio"
              value="shortness"
            />
            <label
              class="text-sm font-medium text-gray-900 dark:text-white"
              data-testid="flowbite-label"
              for="shortness"
            >
              Shortness
            </label>
          </div>
          <div
            class="flex items-center gap-1"
          >
            <input
              class="h-4 w-4 border border-gray-300 text-cyan-600 focus:ring-2 focus:ring-cyan-500 dark:border-gray-600 dark:bg-gray-700 dark:focus:bg-cyan-600 dark:focus:ring-cyan-600"
              id="longness"
              name="sort"
              type="radio"
              value="longness"
            />
            <label
              class="text-sm font-medium text-gray-900 dark:text-white"
              data-testid="flowbite-label"
              for="longness"
            >
              Longness
            </label>
          </div>
        </fieldset>
      </div>
      <div
        class="w-3/6"
      >
        <fieldset
          aria-labelledby="translation-legend"
          class="flex max-w-md flex-rows flex-wrap gap-4"
        >
          <legend
            class="mb-2 dark:text-white"
            id="translation-legend"
          >
            Only examples with translation
          </legend>
          <div
            class="flex items-center gap-1"
          >
            <input
              aria-checked="false"
              class="h-4 w-4 border border-gray-300 text-cyan-600 focus:ring-2 focus:ring-cyan-500 dark:border-gray-600 dark:bg-gray-700 dark:focus:bg-cyan-600 dark:focus:ring-cyan-600"
              id="translation-required"
              name="translation"
              type="radio"
              value="translation-required"
            />
            <label
              class="text-sm font-medium text-gray-900 dark:text-white"
              data-testid="flowbite-label"
              for="translation-required"
            >
              Yes
            </label>
          </div>
          <div
            class="flex items-center gap-1"
          >
            <input
              aria-checked="true"
              checked=""
              class="h-4 w-4 border border-gray-300 text-cyan-600 focus:ring-2 focus:ring-cyan-500 dark:border-gray-600 dark:bg-gray-700 dark:focus:bg-cyan-600 dark:focus:ring-cyan-600"
              id="translation-optional"
              name="translation"
              type="radio"
              value="translation-optional"
            />
            <label
              class="text-sm font-medium text-gray-900 dark:text-white"
              data-testid="flowbite-label"
              for="translation-optional"
            >
              No
            </label>
          </div>
        </fieldset>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
